Improve behavior when not all cluster nodes have plugin

The existing implementation assumes that all Openfire cluster nodes have the plugin loaded. This is not necessarily the case, if only because a plugin can be reloaded on a node, causing it to be briefly unavailable.

This issue was originally identified when testing a fix for #155. See https://github.com/igniterealtime/openfire-monitoring-plugin/pull/156#issuecomment-756775012

Exceptions like these can be found in the logs, when this issue occurs:

```
2021.01.08 13:49:15 DEBUG [pool-6-thread-1]: org.jivesoftware.openfire.container.PluginMonitor - Successfully extracted plugin 'monitoring'.
2021.01.08 13:49:15 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.container.PluginManager - Loading plugin 'monitoring'...
2021.01.08 13:49:16 INFO [pool-155-thread-1]: org.jivesoftware.database.SchemaManager - Found old database version 6 for monitoring. Upgrading to version 7...
2021.01.08 13:49:16 INFO [pool-155-thread-1]: org.jivesoftware.database.SchemaManager - Database update successful.
2021.01.08 13:49:16 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.container.PluginServlet - Loading servlet 'org.jivesoftware.openfire.plugin.monitoring.graph_002dviewer_jsp' of plugin 'monitoring'...
2021.01.08 13:49:16 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.container.PluginServlet - Initializing servlet 'org.jivesoftware.openfire.plugin.monitoring.graph_002dviewer_jsp' of plugin 'monitoring'...
2021.01.08 13:49:16 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.container.PluginServlet - Registering servlet 'org.jivesoftware.openfire.plugin.monitoring.graph_002dviewer_jsp' of plugin 'monitoring' URL patterns.
2021.01.08 13:49:16 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.container.PluginServlet - Servlet 'org.jivesoftware.openfire.plugin.monitoring.graph_002dviewer_jsp' of plugin 'monitoring' loaded successfully.
2021.01.08 13:49:16 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.container.PluginServlet - Loading servlet 'org.jivesoftware.openfire.plugin.monitoring.conversation_002dviewer_jsp' of plugin 'monitoring'...
2021.01.08 13:49:16 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.container.PluginServlet - Initializing servlet 'org.jivesoftware.openfire.plugin.monitoring.conversation_002dviewer_jsp' of plugin 'monitoring'...
2021.01.08 13:49:16 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.container.PluginServlet - Registering servlet 'org.jivesoftware.openfire.plugin.monitoring.conversation_002dviewer_jsp' of plugin 'monitoring' URL patterns.
2021.01.08 13:49:16 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.container.PluginServlet - Servlet 'org.jivesoftware.openfire.plugin.monitoring.conversation_002dviewer_jsp' of plugin 'monitoring' loaded successfully.
2021.01.08 13:49:16 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.container.PluginServlet - Loading servlet 'org.jivesoftware.openfire.plugin.monitoring.conversations_jsp' of plugin 'monitoring'...
2021.01.08 13:49:16 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.container.PluginServlet - Initializing servlet 'org.jivesoftware.openfire.plugin.monitoring.conversations_jsp' of plugin 'monitoring'...
2021.01.08 13:49:16 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.container.PluginServlet - Registering servlet 'org.jivesoftware.openfire.plugin.monitoring.conversations_jsp' of plugin 'monitoring' URL patterns.
2021.01.08 13:49:16 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.container.PluginServlet - Servlet 'org.jivesoftware.openfire.plugin.monitoring.conversations_jsp' of plugin 'monitoring' loaded successfully.
2021.01.08 13:49:16 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.container.PluginServlet - Loading servlet 'org.jivesoftware.openfire.plugin.monitoring.archive_002dconversation_002dparticipants_jsp' of plugin 'monitoring'...
2021.01.08 13:49:16 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.container.PluginServlet - Initializing servlet 'org.jivesoftware.openfire.plugin.monitoring.archive_002dconversation_002dparticipants_jsp' of plugin 'monitoring'...
2021.01.08 13:49:16 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.container.PluginServlet - Registering servlet 'org.jivesoftware.openfire.plugin.monitoring.archive_002dconversation_002dparticipants_jsp' of plugin 'monitoring' URL patterns.
2021.01.08 13:49:16 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.container.PluginServlet - Servlet 'org.jivesoftware.openfire.plugin.monitoring.archive_002dconversation_002dparticipants_jsp' of plugin 'monitoring' loaded successfully.
2021.01.08 13:49:16 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.container.PluginServlet - Loading servlet 'org.jivesoftware.openfire.plugin.monitoring.stats_002dreporter_jsp' of plugin 'monitoring'...
2021.01.08 13:49:16 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.container.PluginServlet - Initializing servlet 'org.jivesoftware.openfire.plugin.monitoring.stats_002dreporter_jsp' of plugin 'monitoring'...
2021.01.08 13:49:16 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.container.PluginServlet - Registering servlet 'org.jivesoftware.openfire.plugin.monitoring.stats_002dreporter_jsp' of plugin 'monitoring' URL patterns.
2021.01.08 13:49:16 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.container.PluginServlet - Servlet 'org.jivesoftware.openfire.plugin.monitoring.stats_002dreporter_jsp' of plugin 'monitoring' loaded successfully.
2021.01.08 13:49:16 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.container.PluginServlet - Loading servlet 'org.jivesoftware.openfire.plugin.monitoring.stats_002ddashboard_jsp' of plugin 'monitoring'...
2021.01.08 13:49:16 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.container.PluginServlet - Initializing servlet 'org.jivesoftware.openfire.plugin.monitoring.stats_002ddashboard_jsp' of plugin 'monitoring'...
2021.01.08 13:49:16 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.container.PluginServlet - Registering servlet 'org.jivesoftware.openfire.plugin.monitoring.stats_002ddashboard_jsp' of plugin 'monitoring' URL patterns.
2021.01.08 13:49:16 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.container.PluginServlet - Servlet 'org.jivesoftware.openfire.plugin.monitoring.stats_002ddashboard_jsp' of plugin 'monitoring' loaded successfully.
2021.01.08 13:49:16 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.container.PluginServlet - Loading servlet 'org.jivesoftware.openfire.plugin.monitoring.archive_002dsearch_jsp' of plugin 'monitoring'...
2021.01.08 13:49:16 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.container.PluginServlet - Initializing servlet 'org.jivesoftware.openfire.plugin.monitoring.archive_002dsearch_jsp' of plugin 'monitoring'...
2021.01.08 13:49:16 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.container.PluginServlet - Registering servlet 'org.jivesoftware.openfire.plugin.monitoring.archive_002dsearch_jsp' of plugin 'monitoring' URL patterns.
2021.01.08 13:49:16 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.container.PluginServlet - Servlet 'org.jivesoftware.openfire.plugin.monitoring.archive_002dsearch_jsp' of plugin 'monitoring' loaded successfully.
2021.01.08 13:49:16 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.container.PluginServlet - Loading servlet 'org.jivesoftware.openfire.plugin.monitoring.archiving_002dsettings_jsp' of plugin 'monitoring'...
2021.01.08 13:49:16 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.container.PluginServlet - Initializing servlet 'org.jivesoftware.openfire.plugin.monitoring.archiving_002dsettings_jsp' of plugin 'monitoring'...
2021.01.08 13:49:16 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.container.PluginServlet - Registering servlet 'org.jivesoftware.openfire.plugin.monitoring.archiving_002dsettings_jsp' of plugin 'monitoring' URL patterns.
2021.01.08 13:49:16 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.container.PluginServlet - Servlet 'org.jivesoftware.openfire.plugin.monitoring.archiving_002dsettings_jsp' of plugin 'monitoring' loaded successfully.
2021.01.08 13:49:16 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.container.PluginServlet - Loading servlet 'JerseyWrapper' of plugin 'monitoring'...
2021.01.08 13:49:16 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.container.PluginServlet - Initializing servlet 'JerseyWrapper' of plugin 'monitoring'...
2021.01.08 13:49:18 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.container.PluginServlet - Registering servlet 'JerseyWrapper' of plugin 'monitoring' URL patterns.
2021.01.08 13:49:18 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.container.PluginServlet - Servlet 'JerseyWrapper' of plugin 'monitoring' loaded successfully.
2021.01.08 13:49:18 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.container.PluginServlet - Loading servlet 'GraphServlet' of plugin 'monitoring'...
2021.01.08 13:49:19 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.container.PluginServlet - Initializing servlet 'GraphServlet' of plugin 'monitoring'...
2021.01.08 13:49:19 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.container.PluginServlet - Registering servlet 'GraphServlet' of plugin 'monitoring' URL patterns.
2021.01.08 13:49:19 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.container.PluginServlet - Servlet 'GraphServlet' of plugin 'monitoring' loaded successfully.
2021.01.08 13:49:19 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.container.PluginServlet - Loading servlet 'ConversationServlet' of plugin 'monitoring'...
2021.01.08 13:49:19 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.container.PluginServlet - Initializing servlet 'ConversationServlet' of plugin 'monitoring'...
2021.01.08 13:49:19 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.container.PluginServlet - Registering servlet 'ConversationServlet' of plugin 'monitoring' URL patterns.
2021.01.08 13:49:19 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.container.PluginServlet - Servlet 'ConversationServlet' of plugin 'monitoring' loaded successfully.
2021.01.08 13:49:19 DEBUG [pool-monitoring2]: com.reucon.openfire.plugin.archive.impl.DatabaseUpdateSplitJIDsTask - No need to run database table migration to split full JID values into bare JID and resource-part components: configuration indicates that migration already occurred.
2021.01.08 13:49:19 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.plugin.MonitoringPlugin - Excluding all public resources from the Authorization-Check filter.
2021.01.08 13:49:19 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.plugin.MonitoringPlugin - Adding the public web sources to the same context as the one that's providing the BOSH interface.
2021.01.08 13:49:19 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.plugin.MonitoringPlugin - Ensure the JSP engine is initialized correctly (in order to be able to cope with Tomcat/Jasper precompiled JSPs).
2021.01.08 13:49:19 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.plugin.MonitoringPlugin - Registering public web context with the embedded webserver.
2021.01.08 13:49:19 DEBUG [pool-155-thread-1]: org.apache.jasper.compiler.JspRuntimeContext - Parent class loader is: [org.jivesoftware.openfire.container.PluginClassLoader@b9442fa]
2021.01.08 13:49:19 DEBUG [pool-155-thread-1]: org.apache.jasper.compiler.JspRuntimeContext - Compilation classpath initialized: /usr/local/openfire/plugins/monitoring/classes/:/usr/local/openfire/plugins/monitoring/database/:/usr/local/openfire/plugins/monitoring/i18n/:/usr/local/openfire/plugins/monitoring/web/:/tmp/jetty-0_0_0_0-7070-classes-_monitoring-any-7081121902843690896/jsp:/usr/local/openfire/plugins/monitoring/classes:/usr/local/openfire/plugins/monitoring/database:/usr/local/openfire/plugins/monitoring/i18n:/usr/local/openfire/plugins/monitoring/web
2021.01.08 13:49:19 DEBUG [pool-155-thread-1]: org.apache.jasper.servlet.JspServlet - Scratch dir for the JSP engine is: [/tmp/jetty-0_0_0_0-7070-classes-_monitoring-any-7081121902843690896/jsp]
2021.01.08 13:49:19 DEBUG [pool-155-thread-1]: org.apache.jasper.servlet.JspServlet - IMPORTANT: Do not modify the generated servlets
2021.01.08 13:49:19 DEBUG [hz.openfire.generic-operation.thread-1]: com.hazelcast.internal.cluster.impl.ClusterHeartbeatManager - [172.60.0.10]:5701 [openfire] [3.12.5] Received heartbeat from Member [172.60.0.20]:5701 - e1d05dd7-9dc0-499c-855b-4a36e327a8c4 (now: 2021-01-08 13:49:19.282, timestamp: 2021-01-08 13:49:19.280)
2021.01.08 13:49:19 DEBUG [archive-monitoring-service-worker-2]: org.jivesoftware.openfire.archive.Archiver - Running with max work queue size 500, max purge interval PT1S, grace period PT0.05S.
2021.01.08 13:49:19 DEBUG [archive-monitoring-service-worker-3]: org.jivesoftware.openfire.archive.Archiver - Running with max work queue size 500, max purge interval PT1S, grace period PT0.05S.
2021.01.08 13:49:19 DEBUG [archive-monitoring-service-worker-4]: org.jivesoftware.openfire.archive.Archiver - Running with max work queue size 500, max purge interval PT1S, grace period PT0.05S.
2021.01.08 13:49:19 DEBUG [pool-monitoring2]: org.jivesoftware.openfire.plugin.util.cache.ClusteredCacheFactory - Executing MultiTask: org.jivesoftware.openfire.reporting.stats.GetStatistics
2021.01.08 13:49:19 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.archive.ArchiveIndexer[CONVERSATION] - Starting...
2021.01.08 13:49:19 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.archive.ArchiveIndexer[CONVERSATION] - Last modification date: null
2021.01.08 13:49:19 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.archive.ArchiveIndexer[CONVERSATION] - Unable to parse 'last modification date' value. Returning EPOCH.
2021.01.08 13:49:19 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.archive.ArchiveIndexer[MUCSEARCH] - Starting...
2021.01.08 13:49:19 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.archive.ArchiveIndexer[MUCSEARCH] - Last modification date: null
2021.01.08 13:49:19 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.archive.ArchiveIndexer[MUCSEARCH] - Unable to parse 'last modification date' value. Returning EPOCH.
2021.01.08 13:49:19 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.archive.ArchiveIndexer[MESSAGE] - Starting...
2021.01.08 13:49:19 DEBUG [pool-monitoring4]: org.jivesoftware.openfire.archive.ArchiveIndexer[CONVERSATION] - Rebuilding the Lucene index...
2021.01.08 13:49:19 DEBUG [pool-monitoring4]: org.jivesoftware.openfire.archive.ArchiveIndexer[CONVERSATION] - Removing old data from directory: /usr/local/openfire/monitoring/search
2021.01.08 13:49:19 DEBUG [pool-monitoring3]: org.jivesoftware.openfire.archive.ArchiveIndexer[MUCSEARCH] - Rebuilding the Lucene index...
2021.01.08 13:49:19 DEBUG [hz.openfire.cached.thread-3]: com.hazelcast.internal.partition.InternalPartitionService - [172.60.0.10]:5701 [openfire] [3.12.5] Checking partition state, version: 678
2021.01.08 13:49:19 DEBUG [pool-monitoring3]: org.jivesoftware.openfire.archive.ArchiveIndexer[MUCSEARCH] - Removing old data from directory: /usr/local/openfire/monitoring/mucsearch
2021.01.08 13:49:19 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.archive.ArchiveIndexer[MESSAGE] - Last modification date: null
2021.01.08 13:49:19 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.archive.ArchiveIndexer[MESSAGE] - Unable to parse 'last modification date' value. Returning EPOCH.
2021.01.08 13:49:19 DEBUG [pool-155-thread-1]: org.jivesoftware.openfire.container.PluginManager - Initialized plugin 'monitoring'.
2021.01.08 13:49:19 DEBUG [pool-monitoring6]: org.jivesoftware.openfire.archive.ArchiveIndexer[MESSAGE] - Rebuilding the Lucene index...
2021.01.08 13:49:19 DEBUG [pool-monitoring6]: org.jivesoftware.openfire.archive.ArchiveIndexer[MESSAGE] - Removing old data from directory: /usr/local/openfire/monitoring/msgsearch
2021.01.08 13:49:19 DEBUG [hz.openfire.InvocationMonitorThread]: com.hazelcast.spi.impl.operationservice.impl.InvocationMonitor - [172.60.0.10]:5701 [openfire] [3.12.5] Invocations:1 timeouts:0 backup-timeouts:0
2021.01.08 13:49:19 INFO [pool-155-thread-1]: org.jivesoftware.openfire.container.PluginManager - Successfully loaded plugin 'monitoring'.
2021.01.08 13:49:19 INFO [pool-6-thread-1]: org.jivesoftware.openfire.container.PluginMonitor - Finished processing all plugins.
2021.01.08 13:49:19 ERROR [pool-monitoring2]: org.jivesoftware.openfire.plugin.util.cache.ClusteredCacheFactory - Failed to execute cluster task
java.util.concurrent.ExecutionException: com.hazelcast.nio.serialization.HazelcastSerializationException: java.lang.ClassNotFoundException: org.jivesoftware.openfire.reporting.stats.GetStatistics
at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.resolveAndThrowIfException(InvocationFuture.java:92) ~[hazelcast-3.12.5.jar!/:?]
at com.hazelcast.spi.impl.AbstractInvocationFuture.get(AbstractInvocationFuture.java:191) ~[hazelcast-3.12.5.jar!/:?]
at com.hazelcast.util.executor.DelegatingFuture.get(DelegatingFuture.java:88) ~[hazelcast-3.12.5.jar!/:?]
at org.jivesoftware.openfire.plugin.util.cache.ClusteredCacheFactory.doSynchronousClusterTask(ClusteredCacheFactory.java:427) [hazelcast-2.5.0.jar!/:?]
at org.jivesoftware.util.cache.CacheFactory.doSynchronousClusterTask(CacheFactory.java:722) [xmppserver-4.6.1.jar:4.6.1]
at org.jivesoftware.openfire.reporting.stats.StatsEngine$SampleTask.run(StatsEngine.java:327) [monitoring-2.2.1-SNAPSHOT.jar!/:?]
at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_275]
at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_275]
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_275]
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_275]
at java.lang.Thread.run(Thread.java:748) [?:1.8.0_275]
Caused by: com.hazelcast.nio.serialization.HazelcastSerializationException: java.lang.ClassNotFoundException: org.jivesoftware.openfire.reporting.stats.GetStatistics
at com.hazelcast.internal.serialization.impl.JavaDefaultSerializers$JavaSerializer.read(JavaDefaultSerializers.java:88) ~[hazelcast-3.12.5.jar!/:?]
at com.hazelcast.internal.serialization.impl.JavaDefaultSerializers$JavaSerializer.read(JavaDefaultSerializers.java:77) ~[hazelcast-3.12.5.jar!/:?]
at com.hazelcast.internal.serialization.impl.StreamSerializerAdapter.read(StreamSerializerAdapter.java:48) ~[hazelcast-3.12.5.jar!/:?]
at com.hazelcast.internal.serialization.impl.AbstractSerializationService.toObject(AbstractSerializationService.java:187) ~[hazelcast-3.12.5.jar!/:?]
at com.hazelcast.executor.impl.operations.AbstractCallableTaskOperation$OffloadImpl.loadCallable(AbstractCallableTaskOperation.java:106) ~[hazelcast-3.12.5.jar!/:?]
at com.hazelcast.executor.impl.operations.AbstractCallableTaskOperation$OffloadImpl.start(AbstractCallableTaskOperation.java:98) ~[hazelcast-3.12.5.jar!/:?]
at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.call(OperationRunnerImpl.java:224) ~[hazelcast-3.12.5.jar!/:?]
at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:199) ~[hazelcast-3.12.5.jar!/:?]
at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:416) ~[hazelcast-3.12.5.jar!/:?]
at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:153) ~[hazelcast-3.12.5.jar!/:?]
at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:123) ~[hazelcast-3.12.5.jar!/:?]
at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.run(OperationThread.java:110) ~[hazelcast-3.12.5.jar!/:?]
at ------ submitted from ------.(Unknown Source) ~[?:?]
at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.resolve(InvocationFuture.java:126) ~[hazelcast-3.12.5.jar!/:?]
at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.resolveAndThrowIfException(InvocationFuture.java:79) ~[hazelcast-3.12.5.jar!/:?]
... 10 more
Caused by: java.lang.ClassNotFoundException: org.jivesoftware.openfire.reporting.stats.GetStatistics
at org.jivesoftware.openfire.plugin.util.cache.ClusterClassLoader.lambda$loadClass$2(ClusterClassLoader.java:94) ~[hazelcast-2.5.0.jar!/:?]
at java.util.Optional.orElseThrow(Optional.java:290) ~[?:1.8.0_275]
at org.jivesoftware.openfire.plugin.util.cache.ClusterClassLoader.loadClass(ClusterClassLoader.java:94) ~[hazelcast-2.5.0.jar!/:?]
at com.hazelcast.nio.ClassLoaderUtil.tryLoadClass(ClassLoaderUtil.java:288) ~[hazelcast-3.12.5.jar!/:?]
at com.hazelcast.nio.ClassLoaderUtil.loadClass(ClassLoaderUtil.java:252) ~[hazelcast-3.12.5.jar!/:?]
at com.hazelcast.nio.IOUtil$ClassLoaderAwareObjectInputStream.resolveClass(IOUtil.java:699) ~[hazelcast-3.12.5.jar!/:?]
at java.io.ObjectInputStream.readNonProxyDesc(ObjectInputStream.java:1986) ~[?:1.8.0_275]
at java.io.ObjectInputStream.readClassDesc(ObjectInputStream.java:1850) ~[?:1.8.0_275]
at java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2160) ~[?:1.8.0_275]
at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1667) ~[?:1.8.0_275]
at java.io.ObjectInputStream.defaultReadFields(ObjectInputStream.java:2405) ~[?:1.8.0_275]
at java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:2329) ~[?:1.8.0_275]
at java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2187) ~[?:1.8.0_275]
at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1667) ~[?:1.8.0_275]
at java.io.ObjectInputStream.readObject(ObjectInputStream.java:503) ~[?:1.8.0_275]
at java.io.ObjectInputStream.readObject(ObjectInputStream.java:461) ~[?:1.8.0_275]
at com.hazelcast.internal.serialization.impl.JavaDefaultSerializers$JavaSerializer.read(JavaDefaultSerializers.java:84) ~[hazelcast-3.12.5.jar!/:?]
at com.hazelcast.internal.serialization.impl.JavaDefaultSerializers$JavaSerializer.read(JavaDefaultSerializers.java:77) ~[hazelcast-3.12.5.jar!/:?]
at com.hazelcast.internal.serialization.impl.StreamSerializerAdapter.read(StreamSerializerAdapter.java:48) ~[hazelcast-3.12.5.jar!/:?]
at com.hazelcast.internal.serialization.impl.AbstractSerializationService.toObject(AbstractSerializationService.java:187) ~[hazelcast-3.12.5.jar!/:?]
at com.hazelcast.executor.impl.operations.AbstractCallableTaskOperation$OffloadImpl.loadCallable(AbstractCallableTaskOperation.java:106) ~[hazelcast-3.12.5.jar!/:?]
at com.hazelcast.executor.impl.operations.AbstractCallableTaskOperation$OffloadImpl.start(AbstractCallableTaskOperation.java:98) ~[hazelcast-3.12.5.jar!/:?]
at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.call(OperationRunnerImpl.java:224) ~[hazelcast-3.12.5.jar!/:?]
at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:199) ~[hazelcast-3.12.5.jar!/:?]
at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:416) ~[hazelcast-3.12.5.jar!/:?]
at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:153) ~[hazelcast-3.12.5.jar!/:?]
at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:123) ~[hazelcast-3.12.5.jar!/:?]
at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.run(OperationThread.java:110) ~[hazelcast-3.12.5.jar!/:?]
2021.01.08 13:49:19 DEBUG [pool-monitoring2]: org.jivesoftware.openfire.plugin.util.cache.ClusteredCacheFactory - Executing MultiTask: org.jivesoftware.openfire.session.GetSessionsCountTask
2021.01.08 13:49:20 DEBUG [pool-monitoring2]: org.jivesoftware.openfire.plugin.util.cache.ClusteredCacheFactory - Executing MultiTask: org.jivesoftware.openfire.muc.cluster.GetNumberConnectedUsers
```

--- a/changelog.html
+++ b/changelog.html
@@ -47,6 +47,7 @@ Monitoring Plugin Changelog
 <p><b>2.2.2</b> -- (tbd)</p>
 <ul>
     <li>[<a href='https://github.com/igniterealtime/openfire-monitoring-plugin/issues/125'>Issue #125</a>] - Combining keyword and date range makes search fail</li>
+    <li>[<a href='https://github.com/igniterealtime/openfire-monitoring-plugin/issues/162'>Issue #162</a>] - Admin console pages should not break when not all cluster nodes have (the same) version of the plugin loaded</li>
     <li>[<a href='https://github.com/igniterealtime/openfire-monitoring-plugin/issues/163'>Issue #163</a>] - Migrate Jive Globals to System Properties</li>
     <li>[<a href='https://github.com/igniterealtime/openfire-monitoring-plugin/issues/190'>Issue #190</a>] - Allow code-update to force a reindexation</li>
     <li>[<a href='https://github.com/igniterealtime/openfire-monitoring-plugin/issues/192'>Issue #192</a>] - Combining keyword and participant(s) makes search fail</li>

--- a/plugin.xml
+++ b/plugin.xml
@@ -6,7 +6,7 @@
     <description>Monitors conversations and statistics of the server.</description>
     <author>Ignite Realtime</author>
     <version>${project.version}</version>
-    <date>12/08/2021</date>
+    <date>12/23/2021</date>
     <minServerVersion>4.7.0</minServerVersion>
     <minJavaVersion>1.8</minJavaVersion>
     <databaseKey>monitoring</databaseKey>

--- a/pom.xml
+++ b/pom.xml
@@ -4,7 +4,7 @@
     <parent>
         <artifactId>plugins</artifactId>
         <groupId>org.igniterealtime.openfire</groupId>
-        <version>4.7.0-beta</version>
+        <version>4.7.0-SNAPSHOT</version>
     </parent>
     <groupId>org.igniterealtime.openfire.plugins</groupId>
     <artifactId>monitoring</artifactId>

--- a/src/i18n/monitoring_i18n.properties
+++ b/src/i18n/monitoring_i18n.properties
@@ -244,5 +244,6 @@ muc.conversation.joined={0} ({1}) has joined the room
 muc.conversation.left={0} ({1}) has left the room
 
 warning.httpbinding.disabled=The HTTP Binding service appears to be disabled! Web-based logs will not be accessible without this service. {0}Please enable the HTTP Binding service here!{1}
+warning.clustering.versions=This Openfire server is part of a cluster. Not all servers in the cluster are running (the same version of) the Monitoring plugin. This will result in inconsistent behavior, possibly even errors. Please review {0}the version of the Monitoring plugin that is installed on each cluster node.{1}
 
 monitoring.search.allow-unrecognized-fields=If 'true', silently ignores unrecognized search filters in queries, which otherwise result in error responses.

--- a/src/java/org/jivesoftware/openfire/archive/ConversationManager.java
+++ b/src/java/org/jivesoftware/openfire/archive/ConversationManager.java
@@ -570,7 +570,13 @@ public class ConversationManager implements ComponentEventListener{
         if (ClusterManager.isSeniorClusterMember()) {
             return conversations.size();
         }
-        return CacheFactory.doSynchronousClusterTask(new GetConversationCountTask(), ClusterManager.getSeniorClusterMember().toByteArray());
+        final Integer count = CacheFactory.doSynchronousClusterTask(new GetConversationCountTask(), ClusterManager.getSeniorClusterMember().toByteArray());
+        if (count == null) {
+            Log.warn("Unable to obtain conversation count from senior cluster member. Is (the same version of) the Monitoring Plugin running there?");
+            return -1;
+        } else {
+            return count;
+        }
     }
 
     /**
@@ -624,14 +630,18 @@ public class ConversationManager implements ComponentEventListener{
             Collection<String> conversationXmls = CacheFactory.doSynchronousClusterTask(new GetConversationsTask(), ClusterManager
                     .getSeniorClusterMember().toByteArray());
             Collection<Conversation> result = new ArrayList<>();
-            for (String conversationXml : conversationXmls) {
-                try {
-                    Log.debug("Interpreting conversation from: {}", conversationXml);
-                    final Conversation conversation = Conversation.fromXml(conversationXml);
-                    Log.debug("Interpreted conversation: {}", conversation);
-                    result.add(conversation);
-                } catch (IOException e) {
-                    Log.warn("Conversation could not be reconstructed from '{}' because of '{}'. This conversation is not included in the result set.", conversationXml, e.getMessage());
+            if (conversationXmls == null) {
+                Log.warn("Unable to obtain conversations from senior cluster member. Is (the same version of) the Monitoring Plugin running there?");
+            } else {
+                for (String conversationXml : conversationXmls) {
+                    try {
+                        Log.debug("Interpreting conversation from: {}", conversationXml);
+                        final Conversation conversation = Conversation.fromXml(conversationXml);
+                        Log.debug("Interpreted conversation: {}", conversation);
+                        result.add(conversation);
+                    } catch (IOException e) {
+                        Log.warn("Conversation could not be reconstructed from '{}' because of '{}'. This conversation is not included in the result set.", conversationXml, e.getMessage());
+                    }
                 }
             }
             return result;

--- a/src/web/archive-search.jsp
+++ b/src/web/archive-search.jsp
@@ -1,9 +1,5 @@
 <%@ page contentType="text/html; charset=UTF-8" %>
 <%@ page errorPage="/error.jsp" import="org.jivesoftware.openfire.plugin.MonitoringPlugin"%>
-<%@ page import="org.jivesoftware.openfire.archive.ArchiveSearch" %>
-<%@ page import="org.jivesoftware.openfire.archive.ArchiveSearcher" %>
-<%@ page import="org.jivesoftware.openfire.archive.Conversation" %>
-<%@ page import="org.jivesoftware.openfire.archive.ConversationManager" %>
 <%@ page import="org.jivesoftware.openfire.XMPPServer" %>
 <%@ page import="org.jivesoftware.openfire.user.UserNameManager" %>
 <%@ page import="org.jivesoftware.openfire.user.UserNotFoundException" %>
@@ -15,6 +11,8 @@
 <%@ page import="org.slf4j.Logger" %>
 <%@ page import="org.slf4j.LoggerFactory" %>
 <%@ page import="com.reucon.openfire.plugin.archive.xep.AbstractXepSupport" %>
+<%@ page import="org.jivesoftware.openfire.archive.*" %>
+<%@ page import="org.jivesoftware.openfire.cluster.ClusterManager" %>
 
 <%@ taglib uri="http://java.sun.com/jsp/jstl/core" prefix="c" %>
 <%@ taglib uri="http://java.sun.com/jsp/jstl/fmt" prefix="fmt" %>
@@ -498,6 +496,16 @@
 </style>
 </head>
 <body>
+
+<% if (!ClusterManager.findRemotePluginsWithDifferentVersion(MonitoringConstants.PLUGIN_NAME).isEmpty()) { %>
+<div class="warning">
+    <fmt:message key="warning.clustering.versions">
+        <fmt:param value="<a href='/system-clustering.jsp'>" />
+        <fmt:param value="</a>" />
+    </fmt:message>
+</div>
+<br/>
+<% } %>
 
 <a href="archive-conversation-participants.jsp?conversationID=" id="lbmessage" title="<fmt:message key="archive.group_conversation.participants" />" style="display:none;"></a>
 

--- a/src/web/archiving-settings.jsp
+++ b/src/web/archiving-settings.jsp
@@ -12,9 +12,10 @@
 <%@ page import="java.util.Map" %>
 <%@ page import="com.reucon.openfire.plugin.archive.impl.MucIndexer" %>
 <%@ page import="org.jivesoftware.openfire.plugin.service.LogAPI" %>
-<%@ page import="org.jivesoftware.util.*" %>
 <%@ page import="org.jivesoftware.openfire.http.HttpBindManager" %>
 <%@ page import="com.reucon.openfire.plugin.archive.impl.MessageIndexer" %>
+<%@ page import="org.jivesoftware.openfire.archive.MonitoringConstants" %>
+<%@ page import="org.jivesoftware.openfire.cluster.ClusterManager" %>
 
 <%@ taglib uri="http://java.sun.com/jsp/jstl/core" prefix="c" %>
 <%@ taglib uri="http://java.sun.com/jsp/jstl/fmt" prefix="fmt" %>
@@ -310,11 +311,22 @@
 
 <div class="success" id="rebuild" style="display: <%=rebuildIndex?"block":"none"%>">
     <fmt:message key="archive.settings.rebuild.success"/>
-</div><br/>
+    <br/>
+</div>
 
 <% if (errors.size() > 0) { %>
 <div class="error">
     <%= errorMessage%>
+</div>
+<br/>
+<% } %>
+
+<% if (!ClusterManager.findRemotePluginsWithDifferentVersion(MonitoringConstants.PLUGIN_NAME).isEmpty()) { %>
+<div class="warning">
+    <fmt:message key="warning.clustering.versions">
+        <fmt:param value="<a href='/system-clustering.jsp'>" />
+        <fmt:param value="</a>" />
+    </fmt:message>
 </div>
 <br/>
 <% } %>

--- a/src/web/conversations.jsp
+++ b/src/web/conversations.jsp
@@ -10,6 +10,8 @@
 <%@ page import="org.jivesoftware.openfire.user.UserManager"%>
 <%@ page import="java.net.URLEncoder" %>
 <%@ page import="org.jivesoftware.util.StringUtils" %>
+<%@ page import="org.jivesoftware.openfire.archive.MonitoringConstants" %>
+<%@ page import="org.jivesoftware.openfire.cluster.ClusterManager" %>
 
 <%@ taglib uri="http://java.sun.com/jsp/jstl/core" prefix="c" %>
 <%@ taglib uri="http://java.sun.com/jsp/jstl/fmt" prefix="fmt" %>
@@ -138,6 +140,16 @@ function updateConversations(data) {
 
 //# sourceURL=conversations.jsp
 </script>
+
+<% if (!ClusterManager.findRemotePluginsWithDifferentVersion(MonitoringConstants.PLUGIN_NAME).isEmpty()) { %>
+<div class="warning">
+    <fmt:message key="warning.clustering.versions">
+        <fmt:param value="<a href='/system-clustering.jsp'>" />
+        <fmt:param value="</a>" />
+    </fmt:message>
+</div>
+<br/>
+<% } %>
 
 <!-- <a href="#" onclick="conversationUpdater(); return false;">click me</a> -->
 <p>

--- a/src/web/stats-dashboard.jsp
+++ b/src/web/stats-dashboard.jsp
@@ -16,6 +16,8 @@
 <%@ page import="java.util.*"%>
 <%@ page import="org.jivesoftware.openfire.XMPPServer" %>
 <%@ page import="org.jivesoftware.openfire.plugin.MonitoringPlugin" %>
+<%@ page import="org.jivesoftware.openfire.cluster.ClusterManager" %>
+<%@ page import="org.jivesoftware.openfire.archive.MonitoringConstants" %>
 <%@ taglib uri="http://java.sun.com/jsp/jstl/core" prefix="c" %>
 <%@ taglib uri="http://java.sun.com/jsp/jstl/fmt" prefix="fmt" %>
 
@@ -380,6 +382,17 @@ function createCookie(name,value,days) {
 </script>
 
 <div id="instructions">
+
+    <% if (!ClusterManager.findRemotePluginsWithDifferentVersion(MonitoringConstants.PLUGIN_NAME).isEmpty()) { %>
+    <div class="warning">
+        <fmt:message key="warning.clustering.versions">
+            <fmt:param value="<a href='/system-clustering.jsp'>" />
+            <fmt:param value="</a>" />
+        </fmt:message>
+    </div>
+    <br/>
+    <% } %>
+
     <table width="756" border="0">
     <tr>
         <td width="426">

--- a/src/web/stats-reporter.jsp
+++ b/src/web/stats-reporter.jsp
@@ -11,6 +11,8 @@
 <%@ page import="java.util.List"%>
 <%@ page import="org.jivesoftware.openfire.plugin.MonitoringPlugin" %>
 <%@ page import="org.jivesoftware.openfire.XMPPServer" %>
+<%@ page import="org.jivesoftware.openfire.cluster.ClusterManager" %>
+<%@ page import="org.jivesoftware.openfire.archive.MonitoringConstants" %>
 <%@ taglib uri="http://java.sun.com/jsp/jstl/core" prefix="c" %>
 <%@ taglib uri="http://java.sun.com/jsp/jstl/fmt" prefix="fmt" %>
 
@@ -280,6 +282,16 @@
 </head>
 
 <body>
+
+<% if (!ClusterManager.findRemotePluginsWithDifferentVersion(MonitoringConstants.PLUGIN_NAME).isEmpty()) { %>
+<div class="warning">
+    <fmt:message key="warning.clustering.versions">
+        <fmt:param value="<a href='/system-clustering.jsp'>" />
+        <fmt:param value="</a>" />
+    </fmt:message>
+</div>
+<br/>
+<% } %>
 
 <table cellpadding="0" cellspacing="0" border="0" width="753">
 <tr>

